tests/FsCheck.Test/Gen.fs

```diff
@@ -119,6 +119,13 @@ module Gen =
         Gen.listOfLength length (Gen.constant v)
         |> sample1
         |> ((=) (List.init length (fun _ -> v)))
+
+    [<Property>]
+    let Shuffle (xs:Set<int>) =
+        not xs.IsEmpty ==> lazy
+        Gen.shuffle xs
+        |> sample 10
+        |> List.exists ((<>) (Set.toSeq xs))
```

Comment:
@kurtschelfthout It *might* be possible to strengthen that test, with the risk of turning it into [an erratic one](http://xunitpatterns.com/Erratic%20Test.html), because of potential [false positives](http://xunitpatterns.com/false%20positive.html).

Taking an `NonEmptySet` to get rid of the precondition sounds like a good idea though :smiley: